buidler/contracts/KyberProject.sol

```diff
@@ -19,21 +19,45 @@ contract KyberProject {
     struct Vote {
         string articleName;
         address article;
+        bool archived;
     }
 
     mapping (address => Vote[]) public articlesVotedOn;
 
-    function registerAVote(address userAddress, string memory _articleName, address _articleAddress) public {
-    	require(check_registration[userAddress] == true, "USER MUST REGISTER FIRST");
-        
+    function registerAVote(address _userAddress, string memory _articleName, address _articleAddress) public {
+    	require(check_registration[_userAddress] == true, "USER MUST REGISTER FIRST");
+
         Vote memory v;
         v.articleName = _articleName;
         v.article = _articleAddress;
+        v.archived = true;
 
         articlesVotedOn[msg.sender].push(v);
     }
 
-    //function getRegisteredUser() public view returns(address[] memory) {
-        //return articlesVotedOn;
+    //function getRegisteredUserInfo(address _userAddress) public view returns(address[] memory) {
+        //return articlesVotedOn[_userAddress];
     //}
-}
\ No newline at end of file
+
+
+    function getArticlesVotedOn(address userAddress) public view returns (string memory articleName, address article, bool archived) {
+       require(check_registration[userAddress] == true, "USER MUST REGISTER FIRST");
```

Comment:
This check might not be required since the user invoking the function need not be a registered user to view the articles voted on for the user address that the function receives